### PR TITLE
filter patient search form on change

### LINKED ISSUE #3711: Remove search button from patient selection form
## Is your feature request related to a problem? Please describe.

With the on-screen keyboard, the search button is hidden and there's only one field showing.
Given that we are only searching local users, switching to a debounced live filter should be responsive enough and would remove the need for the search button.

## Describe the solution you'd like

Remove the search button
Search onChange of any of the fields on the form, with a suitable debounce

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #3711 

## Change summary

Removes the search button from the patient form again.. adds in a debounced onChange filter instead

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Search for patients in step #1
- [ ] Lookup patients in normal dispensing to check nothing else is broken

### Related areas to think about

The debounce on the date field is making it s-u-p-e-r-s-l-o-w when typing. Still works though, and that's outside of these changes, so won't address it here.

### CODE REVIEW

File: src/utilities/constants.js [hunk 1]
```diff
@@ -15,3 +15,7 @@ export const MILLISECONDS = {
 export const SECONDS = {
   ONE_MINUTE: 60,
 };
+
+export const DATE_FORMAT = {
```
Comment (mark-prins) [inline src/utilities/constants.js:19]: Completely not necessary.. it was troubling me slightly that we had a string value all over the place for this one.

Comment (joshxg) [reply src/utilities/constants.js:19]: I like it :)

File: src/widgets/Tabs/PatientSelect.js [hunk 2]
```diff
@@ -59,10 +61,16 @@ const PatientSelectComponent = ({
   onFilterData,
   patients,
   selectPatient,
+  updateForm,
 }) => {
   const columns = React.useMemo(() => getColumns(MODALS.PATIENT_LOOKUP), []);
   const { pageTopViewContainer } = globalStyles;
   const keyboardIsOpen = useKeyboardIsOpen();
+  const debouncedFilter = useDebounce(onFilterData, 500);
```
Comment (joshxg) [inline src/widgets/Tabs/PatientSelect.js:69]: I tried this with `250` and for me it works fine and has a little less lag. Totally dependent on the typist though and if it triggers to much it does get a bit janky (50 is too low!!). 500 is probably good

File: src/widgets/FormControl.js [hunk 3]
```diff
@@ -271,9 +272,11 @@ const mergeProps = (stateProps, dispatchProps, ownProps) => {
     cancelButtonText,
     onCancel,
     canSave = true, // if not specified by calling component, set default
+    onUpdate,
   } = ownProps;
   const onInitialiseForm = () => initialiseForm(inputConfig);
-  const onUpdateForm = (key, value) => !isDisabled && updateForm(key, value);
+  const onUpdateForm = (key, value) =>
```
Comment (joshxg) [inline src/widgets/FormControl.js:278]: I would've thought it doesn't matter what this fn is if the form is disabled, but it does have me curious if there was a reason it was not calling `updateForm` when it was disabled.

Currently if there is an `onUpdate` passed and it is disabled then `updateForm` is set- probably nothing 🤔

**Review by joshxg — APPROVED** [review]
Nice - working excellently. I was wanting some little indication that things had registered and it was about to do the search, like some sort of spinner, but it's probably just because I know what's going on, where a user likely doesn't.

File: src/widgets/FormControl.js (hunk 3 above)
Comment (mark-prins) [reply src/widgets/FormControl.js:278]: oopsie. 6634c014

File: src/widgets/Tabs/PatientSelect.js (hunk 2 above)
Comment (mark-prins) [reply src/widgets/Tabs/PatientSelect.js:69]: Compromised.. went with `300`  though the `250` was fine for me too 🤷‍♂️

**Pull request merged**